src/workerd/util/sqlite.c++

```diff
@@ -174,16 +181,24 @@ class SqliteCallScope {
     /* SQLITE_MISUSE doesn't put error info on the database object, so check it separately */      \
     KJ_ASSERT(_ec != SQLITE_MISUSE, "SQLite misused: " #code, ##__VA_ARGS__);                      \
     if (_ec == SQLITE_IOERR) sqliteCallScope.rethrowVfsError();                                    \
+    if (isCriticalError(code)) {                                                                   \
+      regulator.onCriticalError(_ec, dbErrorMessage(code, db));                                    \
+      KJ_FAIL_REQUIRE("SQLite faield with critical error", dbErrorMessage(_ec, db), ##__VA_ARGS__);\
```

Comment:
No need to throw a unique message here, just let the flow continue and throw the exception it would have thrown otherwise.